app/docs/dev/policies/meshretry.md

```diff
@@ -0,0 +1,293 @@
+---
+title: MeshRetry (beta)
+---
+
+{% warning %}
+This policy uses new policy matching algorithm and is in beta state,
+it should not be mixed with [Retry](../retry).
+{% endwarning %}
+
+This policy enables {{site.mesh_product_name}} to know how to behave if there is a failed scenario (i.e. HTTP request) which could be retried.
+
+## TargetRef support matrix
+
+| TargetRef type    | top level | to  | from |
+| ----------------- | --------- | --- | ---- |
+| Mesh              | ✅        | ✅  | ❌   |
+| MeshSubset        | ✅        | ❌  | ❌   |
+| MeshService       | ✅        | ✅  | ❌   |
+| MeshServiceSubset | ✅        | ❌  | ❌   |
+| MeshGatewayRoute  | ❌        | ❌  | ❌   |
+
+To learn more about the information in this table, see the [matching docs](/docs/{{ page.version }}/policies/targetref).
+
+## Configuration
+
+The policy let you configure retry behaviour for `HTTP`, `GRPC` and `TCP` protocols.
+The protocol is selected by picking the most [specific protocol](/docs/{{ page.version }}/policies/protocol-support-in-kuma/#protocol-support-in-kuma).
+
+Each protocol has a separate section under `default` in the policy YAML.
+Some sections are common between protocols or have similar meaning.
+
+### Retry on
+
+The field `retryOn` is a list of conditions which will cause a retry.
+
+For `HTTP` these are related to the response status code or method ("5xx", "429", "HttpMethodGet").
+For `gRPC` these are status codes in response headers ("canceled", "deadline-exceeded", etc.).
+There is no equivalent for `TCP`.
+
+One or more conditions can be specified, for example:
+
+```yaml
+retryOn:
+  - "429"
+  - "503"
+```
+
+means that it the policy will retry on a status code 429 **or** 503.
+
+### Backoff
+
+[//]: # (TODO: should this be required - https://github.com/kumahq/kuma/blob/b0925593253a186766174b66ca437a696a10c89d/pkg/plugins/policies/meshretry/api/v1alpha1/validator.go#L151 ? can't we just use envoy default of 25ms / 250ms?)
```

Comment:
It's a mergeable field, so it can't be required

Reply:
Yeah, what I mean is that you can't just put 

```json
backOff: {}
```

and use Envoy defaults, I'm just wondering why?

Reply:
but TBH I'm not sure if it's Envoy defaults or Envoy defaults